tasks/Task.cpp

```diff
@@ -119,4 +134,63 @@ void Task::cleanupHook()
 void Task::exceptionHook()
 {
     TaskBase::exceptionHook();
-}
\ No newline at end of file
+}
+
+bool Task::processStartStopCommand()
+{
+    bool cmd;
+    if (_control_cmd.read(cmd) == RTT::NoData) {
+        if (m_running) {
+            m_driver->sendControlCommand(CONTROL_CMD_KEEP_ALIVE);
+        }
```

Comment:
The `control_cmd` port receives true or false to respectively start or stop the generator. After getting it running, the component should repeatedly send keep_alive even if there isn't anything on this input port, otherwise we would go back to having to manually send the keep_alive command every time.

Reply:
Maybe you prefer to continuously send true to keep it running?

Reply:
Ah ... misunderstanding here.

`NoData` means nothing has ever been written on the port (or .clear() was called). When you read a port for which there has been some data written, you get OldData (and the value is re-read in the argument)